### PR TITLE
Add support for setExceptionPauseMode

### LINKED ISSUE #37: Add support for setExceptionPauseMode
It doesn't seem like there's a way to call `setExceptionPauseMode` to force breaking on exceptions.

https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#setexceptionpausemode

### PR DESCRIPTION
**Pull request by DanTup**
This adds a method to `Isolate` to set the exception mode and a test for each mode.

Fixes #37.

### CODE REVIEW

**Comment by DanTup** [comment]
(I believe the Travis failure is because the tests are just broken in the latest SDKs. I ran against an older SDK and all is good. I think @kevmoo was setting this up recently and it's still WIP).

**Review by jacob314 — APPROVED** [review]
Please wait for @nex3 to approve as well.

File: lib/src/isolate.dart
Comment (nex3) [inline lib/src/isolate.dart]: Nit: extra newline

File: lib/src/isolate.dart
Comment (nex3) [inline lib/src/isolate.dart]: Nit: extra newline

File: lib/src/isolate.dart
Comment (nex3) [inline lib/src/isolate.dart]: "step type" is wrong here

**Review by nex3 — CHANGES_REQUESTED** [review]
Since you're providing the ability to set this, you should probably add a getter as well.

**Comment by nex3** [comment]
One other thing: please make sure all the APIs you add document the version of the protocol in which they first appear.

**Comment by DanTup** [comment]
I've added the ability to get the pause mode and an additional test. Also added to changelog and updated version (I'll merge this with the other PR if they're landed around the same time).

@nex3 The test for the getter currently calls `toString()` since the `VMExceptionPauseMode` instances are different. Let me know if you'd prefer this was done another way (I was thinking a factory constructor could use the string to return the same instances, or we could override equality - not really sure what the usual done thing is in Dart).

> One other thing: please make sure all the APIs you add document the version of the protocol in which they first appear.

As far as I can tell from the docs, exceptionPauseMode has always been there (it's not in the changelog), though the stuff in the other PR is new, so I'll go back and update that one.

**Comment by DanTup** [comment]
(btw, someone will need to merge this when happy with it as I don't have merge permissions here)

**Comment by DanTup** [comment]
I've rebased on master since the version number I used has been used in `master` now. Lmk if you'd like any further changes.

**Review by jacob314 — APPROVED** [review]

**Pull request merged**
